oauth2client/service_account.py

```diff
@@ -683,3 +665,29 @@ def _create_token(self, additional_claims=None):
         jwt = crypt.make_signed_jwt(self._signer, payload,
                                     key_id=self._private_key_id)
         return jwt.decode('ascii'), expiry
+
+    def apply(self, headers, access_token=None):
+        """Add the authorization to the headers.
+
+        Args:
+            headers: dict, the headers to add the Authorization header to.
+            access_token: An access token to use instead of the credential's
+                current access token.
+        """
+        if not access_token:
+            access_token = self.access_token
```

Comment:
See `before_request` below - if there is no audience, it generates a one-time token using the request uri as the audience.